### ISSUE TITLE
Component secrets loaded from custom secret store are empty

### ISSUE THREAD

**Issue by christcottrell** [dapr/dapr#7036]
## Expected Behavior

What I am doing is adding auth to the secrets manager component for GCP.  I am adding a reference to an additional secrets manager component to get it from environment variables running locally. In non dev, we'll get that from kubernates, but I believe we would see it in any case.  

This is to protect settings in the yaml file for the secrets component to hide private_key and private_key_id.  I will add my yaml file below in the steps to reproduce the problem. I don't know how anyone is using this and setting their private key in this component without seeing the security risk. And I may reference this issue under security issues. 

I expect the yaml component file to pull the secret from the secret reference. And it does for most fields. 

## Actual Behavior

The error that returns when running it is missing property `private_key` in metadata. There is no issue with private_key_id because I can set an actual value and not a reference and everything runs great.

## Steps to Reproduce the Problem

Below is the yaml file I created.. I will shorten it and take out project numbers, etc

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: secret-store
  namespace: default
spec:
  type: secretstores.gcp.secretmanager
  version: v1
  metadata:
  - name: type
    value: "service_account"
  - name: project_id
    value: "2323232...."
  - name: client_email
    value: "somesvc@someproject.iam.gserviceaccount.com"
  - name: client_id
    value: "some client if"
  - name: auth_uri
    value: "https://accounts.google.com/o/oauth2/auth"
  - name: token_uri
    value: "https://oauth2.googleapis.com/token"
  - name: auth_provider_x509_cert_url
    value: "https://www.googleapis.com/oauth2/v1/certs"
  - name: client_x509_cert_url
    value: "https://www.googleapis.com/robot/v1/metadata/x509/blahblahblah.iam.gserviceaccount.com"
  - name: private_key_id
    secretKeyRef:
      name: SecretsPrivateKeyId
      key: SecretsPrivateKeyId
  - name: private_key
    secretKeyRef:
      name: SecretsPrivateKey
      key: SecretsPrivateKey
auth:
  secretStore: secret-of-secrets
```

So not too much interesting above.. Protecting the secrets through another secret store.. It can be whatever, but I am using an environment secret. It would do this regardless.. 

I'll explain my findings.  Handy dandy google search brought me to https://github.com/dapr/components-contrib/blob/master/secretstores/gcp/secretmanager/secretmanager.go where I looked in the code and see this method.

`
func (s *Store) parseSecretManagerMetadata(metadataRaw secretstores.Metadata) (*GcpSecretManagerMetadata, error) {
	meta := GcpSecretManagerMetadata{}
	metadata.DecodeMetadata(metadataRaw.Properties, &meta)

	if meta.Type == "" {
		return nil, fmt.Errorf("missing property `type` in metadata")
	}
	if meta.ProjectID == "" {
		return nil, fmt.Errorf("missing property `project_id` in metadata")
	}
	if meta.PrivateKey == "" {
		return nil, fmt.Errorf("missing property `private_key` in metadata")
	}
	if meta.ClientEmail == "" {
		return nil, fmt.Errorf("missing property `client_email` in metadata")
	}

	return &meta, nil
}
`

Most of that is probably stuff you don't need to protect except that private_key. Where we find this type of guidance.. "When you create a service account, you are provided with a private key that allows your application or VM to authenticate itself with the Google Cloud APIs. The service account private key is a sensitive credential that should be protected from unauthorized access."

So I thought.. hmm.. I won't use secrets anywhere except the project_id.. And if it is this code, we will get this error, which does happen to match the error I get. And voila, the error appears for project_id. Without digging into that code, my guess is this function is checking the values before it sets any secret references. 

Or am I missing something and there are alternate ways to set the private_key?

I am enjoying the product. Just trying to help!

Thanks!
Chris

## Release Note

RELEASE NOTE:

**Comment by christcottrell** [comment]
Yes.. The secrets are stored locally in an environment variable. 

![image](https://github.com/dapr/components-contrib/assets/53835198/584a8060-8bcf-4d25-aff6-163d384665ca)

apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: secret-of-secrets
spec:
  type: secretstores.local.env
  version: v1
  metadata:

And it is finding the value in the environment store because  if I only use the secrets for private_key_id, it works
- name: private_key_id
    secretKeyRef:
      name: SecretsPrivateKeyId
      key: SecretsPrivateKeyId

And I did the same for the project_id field as mentioned before where all the other settings had values and I got the same type of message thrown in the code I mentioned. 

In the code, and I will dig into it myself as well, but do the secrets get pulled prior to that method being called? Is that where the dapr framework isn't handling it? And if the secrets are not retrieved before that method, it would throw this type of error I am seeing. 

Thanks!
